syntax/go.vim

```diff
@@ -366,13 +366,27 @@ if g:go_highlight_build_constraints != 0
   hi def link goPackageComment    Comment
 endif
 
-" :GoSameIds
-hi def goSameId term=bold cterm=bold ctermbg=white ctermfg=black
-
 " :GoCoverage commands
 hi def link goCoverageNormalText Comment
-hi def      goCoverageCovered    ctermfg=green guifg=#A6E22E
-hi def      goCoverageUncover    ctermfg=red guifg=#F92672
+
+function! s:hi()
+  " :GoSameIds
+  if &background == 'dark'
+    hi def goSameId term=bold cterm=bold ctermbg=white ctermfg=black guibg=white guifg=black
+  else
+    hi def goSameId term=bold cterm=bold ctermbg=14 guibg=Cyan
+  endif
+
+  " :GoCoverage commands
+  hi def      goCoverageCovered    ctermfg=green guifg=#A6E22E
+  hi def      goCoverageUncover    ctermfg=red guifg=#F92672
+endfunction
+
+augroup vim-go-hi
+  autocmd!
+  autocmd ColorScheme * call s:hi()
```

Comment:
It's needed because using `:colorscheme foo` will reset all custom highlight groups − see the message above. It's a bit ugly, but as far as I know there's no "clean" way to avoid this...